pynucastro/networks/rate_collection.py

```diff
@@ -848,10 +848,15 @@ def _nucleon_fraction_nse(self, u, u_c, state):
             else:
                 pf = 1.0
 
+            if nuc.spin_states:
+                spin_state = nuc.spin_states
+            else:
+                spin_state = 1.0
```

Comment:
you can replace the default 1.0 by `raise ValueError('The spin of some nuclei of this network are undefined')`